src/game/eop.js

```diff
@@ -116,321 +109,324 @@ export const ElevationOfPrivilege = Game({
-        return INVALID_MOVE;
-      }
-      let card = getDealtCard(G);
-      return {
-        ...G,
-        threat: {
-          ...G.threat,
-          modal: !G.threat.modal,
-          new: true,
-          owner: ctx.playerID,
-          type: G.suit,
-          id: uuidv4(),
-          title: "",
-          severity: "Medium",
-          description: getThreatDescription(card),
-          mitigation: "",
-        },
-      }
-    },
-    toggleModalUpdate: (G, ctx, threat) => {
-      // if the player has passed, they shouldn't be able to toggle the modal
-      let passed = [...G.passed];
-      if (passed.includes(ctx.playerID) || (threat.owner !== ctx.playerID)) {
-        return INVALID_MOVE;
-      }
 
-      return {
-        ...G,
-        threat: {
-          ...G.threat,
-          modal: !G.threat.modal,
-          new: false,
-          id: threat.id,
-          owner: ctx.playerID,
-          title: threat.title,
-          type: threat.type,
-          severity: threat.severity,
-          description: threat.description,
-          mitigation: threat.mitigation,
-        },
-      }
-    },
-    updateThreat: (G, ctx, field, value) => {
-      return {
-        ...G,
-        threat: {
-          ...G.threat,
-          [field]: value,
+  playerView: PlayerView.STRIP_SECRETS,
+  endIf: (G, ctx) => {
+    if (G.round > G.maxRounds) {
+      let scores = [...G.scores];
+      let winner = 0, max = -1;
+      for (let i=0; i<scores.length; i++) {
+        if (scores[i] > max) {
+          winner = i;
+          max = scores[i];
         }
       }
+      return winner;
+    }
+  },
+  turn: {
+    order: {
+      ...TurnOrder.DEFAULT, // Simple clockwise turns
+      first: function first(G, ctx) { // First player, updated
+        return G.lastWinner;          // every phase (start of game)
+      },
     },
-    selectDiagram: (G, ctx, id) => {
-      // if the player has passed, they shouldn't be able to select diagrams
+    endIf: (G, ctx) => {
       let passed = [...G.passed];
-      if (passed.includes(ctx.playerID)) {
-        return INVALID_MOVE;
-      }
-
-      return {
-        ...G,
-        selectedDiagram: id,
-        selectedComponent: "",
-        selectedThreat: "",
-      }
+      if(passed.length >= ctx.numPlayers) {
+        if(G.numCardsPlayed >= ctx.numPlayers) {
+          //end of trick
+          let lastWinner = getWinner(G.suit, G.dealt);
+          return {next: lastWinner};  // choose next player
+        }
+        return true;
+      };
+      return false;
     },
-    selectComponent: (G, ctx, id) => {
-      // if the player has passed, they shouldn't be able to select components
-      let passed = [...G.passed];
-      if (passed.includes(ctx.playerID)) {
-        return INVALID_MOVE;
-      }
-
-      return {
-        ...G,
-        selectedComponent: id,
-        selectedThreat: "",
+    onEnd: onTurnEnd,
```

Comment:
I have the feeling that the definition of `ElevationOfPrivilege` is still a bit "messy". Some function are defined inline and some as dedicated functions.
I think it is good as done in this line:  Have an external function for each these properties. I.e. Could you move all inline function definition to dedicated functions?

Reply:
Done for most of them. I think maybe setup() - the constructor - might be better off kept in eop.js. And in the turnorder section it makes sense to leave it because the function is so tiny. If you think they're better off in another file though I can move them.

```javascript
order: {
  ...TurnOrder.DEFAULT,
  first: (G, ctx) => {    // <===== This function
    return G.lastWinner;
  },
},
```

Reply:
Yes, it's tiny. But I'd prefer to have this done consistently for all functions.

Reply:
I've moved `first` and `setup` into utils.js